core/src/main/java/com/backblaze/b2/client/B2CopyingPartStorer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019, Backblaze Inc. All Rights Reserved.
+ * License https://www.backblaze.com/using_b2_code.html
+ */
+package com.backblaze.b2.client;
+
+import com.backblaze.b2.client.exceptions.B2Exception;
+import com.backblaze.b2.client.structures.B2Part;
+import com.backblaze.b2.util.B2ByteRange;
+
+import java.util.Objects;
+
+/**
+ * This implementation stores a part of a large file by copying
+ * a file that is already stored in B2.
+ */
+public class B2CopyingPartStorer implements B2PartStorer {
+
+    private int partNumber;
```

Comment:
final?  (in other request classes, too?)

Reply:
Done.